src/Database/Persist/Migration/Core.hs

```diff
@@ -147,14 +147,16 @@ validateMigration migration = do
 -- | Run the given migration. After successful completion, saves the migration to the database.
 runMigration :: MonadIO m => MigrateBackend -> MigrateSettings -> Migration -> SqlPersistT m ()
 runMigration backend settings@MigrateSettings{..} migration = do
-  getMigration backend settings migration >>= mapM_ executeSql
-  now <- liftIO getCurrentTime
-  let version = getLatestVersion migration
-  rawExecute "INSERT INTO persistent_migration(version, label, timestamp) VALUES (?, ?, ?)"
-    [ PersistInt64 $ fromIntegral version
-    , PersistText $ Text.pack $ fromMaybe (show version) $ versionToLabel version
-    , PersistUTCTime now
-    ]
+  currVersion <- fromMaybe (-1) <$> getCurrVersion backend
+  let latestVersion = getLatestVersion migration
+  when (currVersion < latestVersion) $ do
```

Comment:
I'm not fond of the `-1` magic number. Can we change this to
```hs
currVersion <- getCurrVersion backend
let latestVersion = ...
case currVersion of
  Just version | version > latestVersion -> return ()
  _ -> do
    ...
```
?

Reply:
That is definitely the better approach. Wish I could tell you why I made the choice I made. But it's been a while. If you wanna make the change in line you can, otherwise I'll eventually get around to it.

Reply:
fixed.